### PR TITLE
CHE-13381 remove checking for a machine recipe fields which is not ma…

### PR DESCRIPTION
**Pull request by olexii4**
### What does this PR do?
Remove checking for a machine recipe fields which is not mandatory now.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/13381

Signed-off-by: Oleksii Orel <oorel@redhat.com>

### CODE REVIEW

**Comment by olexii4** [comment]
ci-test

**Comment by olexii4** [comment]
ci-test

File: dashboard/src/components/api/environment/kubernetes-machine-recipe-parser.ts [hunk 1]
```diff
@@ -150,12 +150,6 @@ export class KubernetesMachineRecipeParser implements IParser {
   }
 
   validate(recipe: ISupportedListItem ): void {
-    if (!recipe || !recipe.kind) {
-      throw new TypeError(`Recipe item should contain a 'kind' section.`);
-    }
-    if (!recipe.apiVersion) {
-      throw new TypeError(`Recipe item should contain 'apiVersion' section`);
-    }
```
Comment (amisevsk) [inline dashboard/src/components/api/environment/kubernetes-machine-recipe-parser.ts:158]: I'm not sure this is accurate -- the kubernetes machine recipe should be valid kubernetes; without `kind` and `apiVersion` fields it is not. 

Regardless, we can weaken validation if necessary.

**Review by amisevsk — COMMENTED** [review]
I'm not clear on what the actual issue is in this case. Looking at the devfile being used (https://raw.githubusercontent.com/eclipse/che/master/dashboard/devfile.yaml) it seems that the recipe is `dockerimage` type, so why is the kubernetes recipe validator being invoked at all?

Where is the kubernetes recipe that is being validated being generated?

File: dashboard/src/components/api/environment/kubernetes-machine-recipe-parser.ts (hunk 1 above)
Comment (amisevsk) [reply dashboard/src/components/api/environment/kubernetes-machine-recipe-parser.ts:158]: To be clear, I want to understand the root cause of the issue being solved before weakening validation.

**Comment by olexii4** [comment]
@amisevsk  After resolving [devfile.yaml](https://raw.githubusercontent.com/eclipse/che/master/dashboard/devfile.yaml), we have the next workspace config:
```
{
  "defaultEnv": "default",
  "environments": {
    "default": {
      "machines": {
        "dash-dev": {
          "attributes": {
            "memoryLimitBytes": "3221225472"
          },
          "servers": {
            "dashboard-dev-server": {
              "attributes": {},
              "port": "3000",
              "path": "/",
              "protocol": "http"
            }
          },
          "volumes": {
            "projects": {
              "path": "/projects"
            }
          },
          "installers": [],
          "env": {}
        }
      },
      "recipe": {
        "type": "kubernetes",
        "content": "apiVersion: v1\nkind: List\nitems:\n - \n  apiVersion: apps/v1\n  kind: Deployment\n  metadata:\n   annotations: {}\n   labels:\n    che.component.name: dash-dev\n   name: dash-dev\n  spec:\n   selector:\n    matchLabels:\n     che.component.name: dash-dev\n   template:\n    metadata:\n     annotations:\n      org.eclipse.che.container.dash-dev.machine_name: dash-dev\n     labels:\n      che.component.name: dash-dev\n     name: dash-dev\n    spec:\n     containers:\n      - \n       image: 'eclipse/che-dashboard-dev:nightly'\n       name: dash-dev\n       resources:\n        limits:\n         memory: 3Gi\n        requests: {}\n     nodeSelector: {}\n",
        "contentType": "application/x-yaml"
      }
    }
  },
  "projects": [
    {
      "links": [],
      "name": "che",
      "attributes": {},
      "source": {
        "location": "https://github.com/eclipse/che.git",
        "type": "git",
        "parameters": {
          "branch": "master"
        }
      },
      "path": "/che",
      "mixins": [],
      "problems": []
    }
  ],
  "name": "dashboard-in-che",
  "attributes": {
    "editor": "eclipse/che-theia/next",
    "editorComponentAlias": "theia-editor",
    "plugins": "che-incubator/typescript/1.30.2,eclipse/che-machine-exec-plugin/0.0.1",
    "pluginComponentsAliases": "che-incubator/typescript/1.30.2=typescript-plugin,eclipse/che-machine-exec-plugin/0.0.1=exec-plugin"
  },
  "commands": [
    {
      "commandLine": "cd /projects/che/dashboard && yarn",
      "name": "dashboard_build",
      "attributes": {
        "componentAlias": "dash-dev",
        "machineName": "dash-dev",
        "workingDir": "/projects/che/dashboard"
      },
      "type": "exec"
    },
    {
      "commandLine": "cd /projects/che/dashboard && yarn test",
      "name": "dashboard_test",
      "attributes": {
        "componentAlias": "dash-dev",
        "machineName": "dash-dev",
        "workingDir": "/projects/che/dashboard"
      },
      "type": "exec"
    },
    {
      "commandLine": "cd /projects/che/dashboard && node_modules/.bin/gulp serve --server=${CHE_API_EXTERNAL%????}",
      "name": "dashboard_dev_server",
      "attributes": {
        "componentAlias": "dash-dev",
        "machineName": "dash-dev",
        "workingDir": "/projects/che/dashboard"
      },
      "type": "exec"
    }
  ],
  "links": []
}
```

 recipeType -  "kubernetes"

File: dashboard/src/components/api/environment/kubernetes-machine-recipe-parser.ts (hunk 1 above)
Comment (olexii4) [reply dashboard/src/components/api/environment/kubernetes-machine-recipe-parser.ts:158]: Machine recipe - client-side abstraction. It is a part of the valid kubernetes recipe (just a container). And we cower them with some information from the parent pod before showing to end user. 

In any case, it is just a part of the valid kubernetes recipe. And we should check this part only.

**Review by amisevsk — APPROVED** [review]

**Review by ashumilova — APPROVED** [review]

**Comment by olexii4** [comment]
ci-test

**Review by akurinnoy — APPROVED** [review]

**Pull request merged**